lib/pbench/test/unit/server/query_apis/test_controllers_list.py

```diff
@@ -121,31 +179,48 @@ def test_query(self, server_config, query_api, user_ok, find_template):
         }
 
         index = self.build_index(server_config, ("2020-08", "2020-09", "2020-10"))
+        # If we're not asking about a particular user, or if the user
+        # field is to be omitted altogether, or if we have a valid
+        # token, then the request should succeed.
+        if not user or user == "no_user" or build_auth_header[1] == "valid":
```

Comment:
Having the `build_auth_header` fixture return a named tuple might make this interface contract just a bit easier to follow. (Or even just a simple `dict`.)